lol_message/blog.md

```diff
@@ -0,0 +1,571 @@
+
+Once you can see the dancing pigeon, you are ready to use Courier.
+
+
+After setting up the Courier account, we will be creating a new notification template in the Courier dashboard.
+
+- Go to the [Courier dashboard](https://app.courier.com/) and click on the `Create Notification` button.
+
+- Select the `Email` channel and click on the `publish changes` button.
+
+- After publishing the changes, click on the email icon from the left sidebar to create a new email template.
+
+Your email template should look like this without logo:
+
+![Template](https://i.imgur.com/RJBR4aQ.png)
+
+Now we will be adding content to the email template.You can use the following format:
+
+![Template 1](https://i.imgur.com/DbhGtt7.png)
+
+![Template 2](https://i.imgur.com/TlZaIbx.png)
+
+
+Small explanation of the above template:
+
+- `{username}` is the username of the user who received the message.
+
+- `{question}` is the question that the user asked.
+
+- `{subject}` is the subject of the message randomly generated.
+
+-  `{url}` is the link to the message page.
+
+Lastly, we need to copy the `template ID` and `Authorization` token from the Courier dashboard.
+
+For that click on the `Send` tab, and you will see the `template ID` and `Authorization` token.
+
+![Send Tab](https://i.imgur.com/HCLeWvn.png)
+
+Copy the template ID and Authorization token and paste it in `.env` file.
+
+Use the following format:
+
+```
+COURIER_API_KEY=YOUR_AUTH_TOKEN
+COURIER_TEMPLATE_ID=YOUR_TEMPLATE_ID
+```
+
+Ok now we are done with the Courier setup. Let's move on to the next part.
+
+### Part 3: Backend
+
+This part is the most important part of the project. We will be building the backend API using [tRPC](https://trpc.io/) and integrating it with [Courier](https://courier.com/) to send email notifications.
+
+*Note*: You can use any backend framework of your choice
+
+
+So let's get started with the backend API.
+
+First, we need to install the `trpc` and configure it with our Next.js project.
+
+Please check out the following documentation to configure `trpc` with Next.js [here](https://trpc.io/docs/v9/nextjs).
+
+
+After successfully configured `trpc` with Next.js, we need to install `Courier` client using the following command:
+
+```bash
+yarn add @trycourier/courier
+```
+
+Ok now we are ready to build the backend API.
+
+First, we need to create a new file `utils/database.ts` and add the following code:
+
+```ts
+import { PrismaClient } from "@prisma/client";
+
+declare global {
+  var prisma: PrismaClient | undefined;
+}
+
+export const database =
+  global.prisma ||
+  new PrismaClient({
+    log: ["query"],
+  });
+
+if (process.env.NODE_ENV !== "production") global.prisma = database;
+```
+
+The above code is used to connect to the database using `prisma` client.
+
+Next open your trpc router file and add the following code:
+
+```ts
```

Comment:
Let's break this down too. It can be a lot for someone who has never done this before.